suite/suite.go

```diff
@@ -55,6 +55,21 @@ func (suite *Suite) Assert() *assert.Assertions {
 	return suite.Assertions
 }
 
+// Run is a helper method to do a simple subtest. It sets `T()` to the
+// `*testing.T` of the subtest.
+func (suite *Suite) Run(name string, f func()) {
+	oldT := suite.T()
+	suite.T().Run(name, func(t *testing.T) {
+		suite.SetT(t)
+
+		defer func() {
+			suite.SetT(oldT)
+		}()
+
+		f()
```

Comment:
Rather than receiving an empty `func()`, maybe we should receive `func(*Suite)` and send a test suite with the new `*testingT`?

Would that prevent race conditions?